src/js/UI.js

```diff
@@ -80,21 +83,43 @@ export default class UI {
         }
     }
 
+    remove_old() {
+        while (this.dom.output.lastChild)
+            this.dom.output.removeChild(this.dom.output.lastChild)
+        while (this.dom.exitcode.lastChild)
+            this.dom.exitcode.removeChild(this.dom.exitcode.lastChild)
+        while (this.dom.errmsg.lastChild)
+            this.dom.errmsg.removeChild(this.dom.errmsg.lastChild)
+        while (this.dom.warnings.lastChild)
+            this.dom.warnings.removeChild(this.dom.warnings.lastChild)
+    }
+
     /**
      * Adds output to DOM.
      * @param {String} code Exit code of program.
      * @param {String} output Console output of program.
+     * @param {String} compiler_output Console output of the compiler.
      * @returns {void}
      */
-    output(code, output) {
-        const content = output ? document.createTextNode(`${output}\n\nProgram exited with code ${code}`) :
-                                 document.createTextNode(`Program exited with code ${code}`)
+    output(code, output, compiler_output) {
+        const content = document.createTextNode(`${output}`)
+        const exitcode = document.createTextNode(`Program exited with code ${code}.`)
+        const warnings = compiler_output
```

Comment:
I think in this case it would be better to not add any node at all as it really doesn't matter cost-wise whether you add empty text node or node with some text(performance wise).